### DISCUSSION TITLE
Synchronisation from outdated project on QField

### DISCUSSION THREAD

**Discussion by sachatk** [opengisch/QField#3909] [Q&A]
Hello, we are working with some people on the field in Africa. They pushed the data they have been sampling today but the synchronization failed . When I try to Re-apply the changes, it does not work (see picture 1). Based on the info I got, I think the people on the field do not have the last version of the project (see picture 2). I would like to resolve this by changing the raw JSON (see picture 3) but cannot do this. Do you have any idea how I can resolve this  ? 
![picture_1](https://user-images.githubusercontent.com/92298302/219063731-37cc165d-2ccc-45a3-914f-97b0e8df42b6.PNG)
![picture_2](https://user-images.githubusercontent.com/92298302/219063741-8c8402e7-69e0-4cda-aff6-972917b2525e.PNG)
![picture_3](https://user-images.githubusercontent.com/92298302/219063748-789b3322-26d3-49a5-a9b5-544bb11f1500.PNG)

**Comment by suricactus** [comment]
Is there any chance the user `nouri` had an outdated version of your project file? Is it possible this layer has been deleted and readded in you project file?

**Reply by sachatk** [reply]
Yess the user Nouri has an outdated version of our project file... He did not synchronize before going to the field. Any ideahow I could solve this issue ? Not a lot has changed between the old and new project file.

**Reply by suricactus** [reply]
Short answer: no, you cannot do much beyond manual fix.

Long answer: you can trick the system in two different ways.

1) download the `.qgs` file from the cloud and manually change the layer id in the XML file to match the one in the error, or just reupload the outdated .qgs file. Do not reupload anything else besides the .qgs file. Then try to reapply the changes, QFieldCloud should accept them.

2) download the raw JSON files on you machine and feed them to the `apply_deltas.py` script https://github.com/opengisch/qfieldcloud/blob/master/docker-qgis/apply_deltas.py. You can get some inspiration how to run it from here https://github.com/opengisch/qfieldcloud/blob/master/docker-qgis/tests/test_apply_deltas.sh .

Either way, it is far from the most pleasent thing to do. We will look for QFieldCloud native solution for such problems in the future, but this is all we have for now.

If you need extra support, please drop us an email on info@opengis.ch .

**Reply by sachatk** [reply]
You mean the qgz file ? I do not have any qgs file. 
![all_files](https://user-images.githubusercontent.com/92298302/219075851-46744aa9-b330-417e-b1d1-c0ec34cd4834.PNG)

**Reply by suricactus** [reply]
.qgz is the zipped version of the .qgs file. If you open the .qgz file with your archive manager you will find your .qgs file inside. So they are practically the same thing, using the .qgz has some benefits from user perspective and it is the default in QGIS.

**Reply by sachatk** [reply]
Hello, could you show me how to access this download this qgs file and acces the XML file ? I cannot find this file... when downloading the project via the QGIS plugin...

**Reply by suricactus** [reply]
Hey @sachatk , as I said, you have to open the .qgz file with your archive manager program, not with QGIS. Note both solutions I have mentioned above are for quite advanced QGIS/Python users. I will leave any further help to the broader QFieldCloud community for now. As I said before, you can also ask for a support offer on info@opengis.ch .
